### ISSUE TITLE
 Investigate and fix issues reported by Coverity Scan

### ISSUE THREAD

**Issue by stweil** [kitodo/kitodo-production#5997]
The static code analyzer Coverity Scan reports 330 open issues (after #5995 is applied) for the current git master branch. A quick review of those issues shows that at least some of them are real bugs which must be fixed.

Therefore all issues should be reviewed and classified (Coverity Scan has classifications for bug, false positive, intentional and pending). Ideally bugs should also be fixed.

**Comment by stweil** [comment]
The estimated costs for the classification of all reported issues are a few days (low).

Fixing bugs ranges from low costs for trivial fixes, but can also require medium or even high costs for more complex bugs. Therefore I'd restrict this task to fixing trivial bugs and reporting all other bugs as issues on GitHub.

Ideally all remaining issues in Coverity Scan which were classified as bugs should contain references to GitHub issues.

**Comment by solth** [comment]
Votes: 8

**Comment by stweil** [comment]
[Coverity Scan status](https://scan.coverity.com/projects/kitodo-kitodo-production) update for Git master branch:
2024-08-05: 393 outstanding issues
2024-08-22: 354 outstanding issues
2025-03-13: 358 outstanding issues
2026-06-24: 296 outstanding issues (with #7094 applied)

All but one resource leaks are fixed in #7094. About 28 issues are null pointer dereferences. 2 issues are known security issues. The rest are medium or low priority issues.

**Comment by stweil** [comment]
@solth, among the low priority issues are more than 30 of this kind:
> PI: Do not reuse public identifiers from Java Standard Library.
> It's essential to avoid reusing public identifiers from the Java Standard Library as class names.
> It is a good practice to avoid reusing public identifiers from the Java Standard Library as field names in your code.
> [...]

Examples of such identifiers are `FileFormat`, `Option`, `Namespace`, all very generic names, so the conflicts are not surprising.

Do we want to ignore such issues, or should we try to use more specific names, perhaps by prepending "Kitodo" to the name?
